starknet/test/l1-execution.test.ts

```diff
@@ -0,0 +1,435 @@
+      abiCoder.encode(
+        ['tuple(address to, uint256 value, bytes data, uint8 operation, uint256 salt)[]'],
+        [[proposalTx]],
+      ),
+    );
+    // Represent the execution hash as a Cairo Uint256
+    const executionHashUint256: Uint256 = uint256.bnToUint256(executionHash);
+
+    const executionPayload = [
+      l1AvatarExecutionStrategy.address,
+      executionHashUint256.low,
+      executionHashUint256.high,
+    ];
+
+    // Propose
+    await account.execute({
+      contractAddress: starkTxAuthenticator.address,
+      entrypoint: 'authenticate_propose',
+      calldata: CallData.compile({
+        space: space.address,
+        author: account.address,
+        executionStrategy: {
+          address: ethRelayer.address,
+          params: executionPayload,
+        },
+        userProposalValidationParams: [],
+        metadataURI: [],
+      }),
+    });
+
+    // Vote
+    await account.execute({
+      contractAddress: starkTxAuthenticator.address,
+      entrypoint: 'authenticate_vote',
+      calldata: CallData.compile({
+        space: space.address,
+        voter: account.address,
+        proposalId: { low: '0x1', high: '0x0' },
+        choice: '0x2',
+        userVotingStrategies: [{ index: '0x0', params: [] }],
+        metadataURI: [],
+      }),
+    });
+
+    // TODO: Advance time so that the maxVotingTimestamp is exceeded
+    await starknet.devnet.increaseTime(20);
+    await sleep(20000);
+
+    // Execute
+    await account.execute({
+      contractAddress: space.address,
+      entrypoint: 'execute',
+      calldata: CallData.compile({
+        proposalId: { low: '0x1', high: '0x0' },
+        executionPayload: executionPayload,
+      }),
+    });
+
+    // Propogating message to L1
+    const flushL2Response = await starknet.devnet.flush();
+    const message_payload = flushL2Response.consumed_messages.from_l2[0].payload;
+    // Proposal data can either be extracted from the message sent to L1 (as done here) or from the pulled from the contract directly
+    const space_message = message_payload[0];
+    const proposal = {
+      startTimestamp: message_payload[1],
+      minEndTimestamp: message_payload[2],
+      maxEndTimestamp: message_payload[3],
+      finalizationStatus: message_payload[4],
+      executionPayloadHash: message_payload[5],
+      executionStrategy: message_payload[6],
+      authorAddressType: message_payload[7],
+      author: message_payload[8],
+      activeVotingStrategies: uint256.uint256ToBN({
+        low: message_payload[9],
+        high: message_payload[10],
+      }),
+    };
+    const forVotes = uint256.uint256ToBN({
+      low: message_payload[11],
+      high: message_payload[12],
+    });
+    const againstVotes = uint256.uint256ToBN({
+      low: message_payload[13],
+      high: message_payload[14],
+    });
+    const abstainVotes = uint256.uint256ToBN({
+      low: message_payload[15],
+      high: message_payload[16],
+    });
```

Comment:
might be worth doing a function `extract_data` that we can call like so 
`(proposal, forVotes, againstvotes, abstainVotes) = extract_data(message_payload)` ?